dm/config/source_config.go

```diff
@@ -270,8 +292,15 @@ func (c *SourceConfig) AdjustServerID(ctx context.Context, db *sql.DB) error {
 
 // LoadFromFile loads config from file.
 func (c *SourceConfig) LoadFromFile(path string) error {
-	metaData, err := toml.DecodeFile(path, c)
-	return c.check(&metaData, err)
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return terror.ErrConfigReadCfgFromFile.Delegate(err, path)
+	}
+	if err := yaml.UnmarshalStrict(content, c); err != nil {
+		return terror.ErrConfigYamlTransform.Delegate(err, "decode source config")
+	}
+	c.adjust()
+	return nil
```

Comment:
how about using `ParseYaml`

Reply:
`(*SourceConfig).check(metaData)` also checked the undecoded items, `UnmarshalStrict` is similar to that behaviour